app/controllers/rooms_controller.rb

```diff
@@ -98,6 +98,9 @@ def update
 
   # POST /:room_uid
   def join
+    # If this setting is turned on only authenticated users are allowed to join rooms
+    return if room_authentication_required && current_user.nil?
```

Comment:
I think it might be a better idea to redirect the user to the sign in page with a flash saying "You must be logged in to join the room"